packages/@aws-cdk/aws-s3objectlambda/lib/access-point.ts

```diff
@@ -0,0 +1,263 @@
+
+  /**
+   * The creation data of the access point.
+   */
+  readonly accessPointCreationDate: string;
+}
+
+/**
+  * An S3 Object Lambda Access Point for intercepting and
+  * transforming `GetObject` requests.
+  */
+export class AccessPoint extends AccessPointBase {
+  /**
+   * Reference an existing AccessPoint defined outside of the CDK code.
+   */
+  public static fromAccessPointAttributes(scope: Construct, id: string, attrs: AccessPointAttributes): IAccessPoint {
+    const arn = core.Arn.split(attrs.accessPointArn, core.ArnFormat.SLASH_RESOURCE_NAME);
+    if (!arn.resourceName) {
+      throw new Error('Unable to parse acess point name');
+    }
+    const name = arn.resourceName;
+    class Import extends AccessPointBase {
+      public readonly accessPointArn: string = attrs.accessPointArn;
+      public readonly accessPointCreationDate: string = attrs.accessPointCreationDate;
+      protected name: string = name;
+    }
+    return new Import(scope, id);
+  }
+
+  private readonly accessPoint: CfnAccessPoint
+  protected readonly name: string
+
+  /**
+   * The ARN of the access point.
+   * @attribute
+   */
+  public readonly accessPointArn: string
+
+  /**
+   * The creation data of the access point.
+   * @attribute
+   */
+  public readonly accessPointCreationDate: string
+
+  constructor(scope: Construct, id: string, props: AccessPointProps) {
+    super(scope, id);
+
+    const supporting = new s3.CfnAccessPoint(this, 'AccessPoint', {
+      bucket: props.bucket.bucketName,
+    });
+    supporting.addPropertyOverride('Name', `${props.accessPointName}-access-point`);
+
+    const allowedFeatures = [];
+    if (props.supportsGetObjectPartNumber) {
+      allowedFeatures.push('GetObject-PartNumber');
+    }
+    if (props.supportsGetObjectRange) {
+      allowedFeatures.push('GetObject-Range');
+    }
+
+    this.name = props.accessPointName.toLowerCase();
+    this.accessPoint = new CfnAccessPoint(this, 'LambdaAccessPoint', {
+      name: this.name,
+      objectLambdaConfiguration: {
+        allowedFeatures,
+        cloudWatchMetricsEnabled: props.cloudWatchMetricsEnabled,
+        supportingAccessPoint: supporting.getAtt('Arn').toString(),
+        transformationConfigurations: [
+          {
+            actions: ['GetObject'],
+            contentTransformation: {
+              AwsLambda: {
+                FunctionArn: props.fn.functionArn,
+                FunctionPayload: props.payload ?? '',
+              },
+            },
+          },
+        ],
+      },
+    });
+    this.accessPoint.addDependsOn(supporting);
+
+    this.accessPointArn = this.accessPoint.attrArn;
+    this.accessPointCreationDate = this.accessPoint.attrCreationDate;
+
+    props.fn.addToRolePolicy(
+      new iam.PolicyStatement({
+        actions: ['s3-object-lambda:WriteGetObjectResponse'],
+        resources: ['*'],
+      }),
+    );
+  }
+
+  /** Implement the {@link IAccessPoint.domainName} field. */
+  get domainName(): string {
```

Comment:
These should either be here or on the abstract class, you don't need them both places.